frontend/javascripts/oxalis/view/action-bar/download_modal_view.tsx

```diff
@@ -0,0 +1,463 @@
+}
+
+export async function copyToClipboard(code: string) {
+  await navigator.clipboard.writeText(code);
+  Toast.success("Snippet copied to clipboard.");
+}
+
+function MoreInfoHint() {
+  return (
+    <Hint
+      style={{
+        margin: "0px 12px 0px 12px",
+      }}
+    >
+      For more information on how to work with annotation files visit the{" "}
+      <a
+        href="https://docs.webknossos.org/webknossos/tooling.html"
+        target="_blank"
+        rel="noreferrer"
+      >
+        user documentation
+      </a>
+      .
+    </Hint>
+  );
+}
+
+function CopyableCodeSnippet({ code, onCopy }: { code: string; onCopy?: () => void }) {
+  return (
+    <pre>
+      <Button
+        style={{
+          float: "right",
+          border: "none",
+          width: "18px",
+          height: "16px",
+          background: "transparent",
+        }}
+        onClick={() => {
+          copyToClipboard(code);
+          if (onCopy) {
+            onCopy();
+          }
+        }}
+        icon={<CopyOutlined />}
+      />
+      {code}
+    </pre>
+  );
+}
+
+const okTextForTab = new Map([
+  ["download", "Download"],
+  ["export", "Start Export Job"],
+  ["python", null],
+]);
+
+function Footer({ tabKey, onClick }: { tabKey: string; onClick: () => void }) {
+  const okText = okTextForTab.get(tabKey);
+  return okText != null ? (
+    <Button
+      key="ok"
+      type="primary"
+      disabled={tabKey === "export" && !features().jobsEnabled}
+      onClick={onClick}
+    >
+      {okText}
+    </Button>
+  ) : null;
+}
+
+export default function DownloadModalView(props: Props): JSX.Element {
+  const { isVisible, onClose, annotationType, annotationId, hasVolumeFallback } = props;
+  const userBoundingBoxes = useSelector((state: OxalisState) =>
+    getUserBoundingBoxesFromState(state),
+  );
+  const [activeTabKey, setActiveTabKey] = useState("download");
+  const [includeVolumeData, setIncludeVolumeData] = useState(true);
+  const [keepWindowOpen, setKeepWindowOpen] = useState(false);
+
+  // const layers = chooseSegmentationLayer ? getSegmentationLayers(dataset) : getColorLayers(dataset);
+  const tracing = useSelector((state: OxalisState) => state.tracing);
+  const dataset = useSelector((state: OxalisState) => state.dataset);
+  const [startedExports, setStartedExports] = useState([]);
+  const layers = getDataLayers(dataset);
+
+  const isMergerModeEnabled = useSelector(
+    // @ts-expect-error ts-migrate(2339) FIXME: Property 'temporaryConfiguration' does not exist o... Remove this comment to see the full error message
+    (state) => state.temporaryConfiguration.isMergerModeEnabled,
+  );
+  const activeMappingInfos = useSelector(
+    // @ts-expect-error ts-migrate(2339) FIXME: Property 'temporaryConfiguration' does not exist o... Remove this comment to see the full error message
+    (state) => state.temporaryConfiguration.activeMappingByLayer,
+  );
+
+  const [selectedLayerName, setSelectedLayerName] = useState("");
```

Comment:
better than defaulting to an empty string would be defaulting to null. then, the typechecker will complain if the null case is not handled somewhere. I'll change this in a commit, since I also saw some other small type issues which are easier to fix than to explain ^^